crypto/dilithium/ml_dsa.c

```diff
@@ -25,34 +26,34 @@
 // depending on platform support.
 
 int ml_dsa_65_keypair(uint8_t *public_key  /* OUT */,
-                       uint8_t *secret_key /* OUT */) {
+                      uint8_t *secret_key  /* OUT */) {
   ml_dsa_params params;
   ml_dsa_65_params_init(&params);
-  return crypto_sign_keypair(&params, public_key, secret_key);
+  return (crypto_sign_keypair(&params, public_key, secret_key) == 0);
```

Comment:
how did this work so far? aren't we checking the return value? because this change inverts the return value

Reply:
`crypto_sign_keypair` returns `0` on success `-1` for failure. This is an anti-pattern with how EVP expects `1` for success. The return values are checked within the `PKEY` functions:

- keygen: https://github.com/jakemas/aws-lc/blob/7435f9b2920753efc71a6461c55be213dc24da65/crypto/dilithium/p_pqdsa.c#L52 prior to this change we check: `ml_dsa_65_keypair(key->pub, key->priv) != 0`. For more consistency, we do this check at the lowest level API, and instead within PKEY keygen we check: `!pqdsa->method->keygen(key->public_key, key->secret_key)`